packages/x-charts-pro/src/internals/plugins/useChartProZoom/gestureHooks/usePanOnDrag.ts

```diff
@@ -23,7 +23,8 @@ export const usePanOnDrag = (
 ) => {
   const drawingArea = useSelector(store, selectorChartDrawingArea);
   const optionsLookup = useSelector(store, selectorChartZoomOptionsLookup);
-  const startRef = React.useRef<readonly ZoomData[]>(null);
+  const isInteracting = React.useRef<boolean>(false);
+  const accumulatedChange = React.useRef<{ x: number; y: number }>({ x: 0, y: 0 });
```

Comment:
I don't think these needs to be refs. Since they're only used inside the `useEffect` we can create `const` variable with interior mutability to achieve the same effect. 

Basically, we create these variables inside the `useEffect` similar to a ref: `const accumulatedChange = { current: { x: 0, y: 0 }}`. 

It would also allow us to unit test this logic, which I think we should because it's prone to errors as explained [here](https://github.com/mui/mui-x/pull/20163/files#r2481935363).